app/services/school_partnerships/create.rb

```diff
@@ -0,0 +1,121 @@
+    end
+
+    def enabled_contract_period_year_exists
+      errors.add(:contract_period_year, "The '#/cohort' you have entered is invalid. Check cohort details and try again.") unless contract_period
+    end
+
+    def lead_provider_exists
+      errors.add(:lead_provider_id, "Enter a '#/lead_provider_id'.") unless lead_provider
+    end
+
+    def school_exists
+      errors.add(:school_api_id, "The '#/school_id' you have entered is invalid. Check school details and try again. Contact the DfE for support if you are unable to find the '#/school_id'.") unless school
+    end
+
+    def school_is_not_cip_only
+      errors.add(:school_api_id, "The school you have entered has not registered to deliver DfE-funded training. Contact the school for more information.") if school&.eligible_for_cip?
+    end
+
+    def school_is_eligible
+      errors.add(:school_api_id, "The school you have entered is currently ineligible for DfE funding. Contact the school for more information.") unless school&.eligible_for_fip?
+    end
+
+    def school_partnership_does_not_already_exists
+      return unless school && active_lead_provider
+
+      existing_school_partnership = school
+        .school_partnerships
+        .joins(:lead_provider_delivery_partnership)
+        .exists?(lead_provider_delivery_partnerships: { active_lead_provider: })
+
+      errors.add(:school_api_id, "You are already in a confirmed partnership with this school for the entered cohort.") if existing_school_partnership
+    end
+
+    def delivery_partner_exists
+      errors.add(:delivery_partner_api_id, "The '#/delivery_partner_id' you have entered is invalid. Check delivery partner details and try again.") unless delivery_partner
+    end
+
+    def active_lead_provider
+      return unless lead_provider && contract_period
+
+      @active_lead_provider ||= lead_provider.active_lead_providers.find_by(contract_period:)
+    end
+
+    def lead_provider_delivery_partnership
+      return unless active_lead_provider && delivery_partner
+
+      @lead_provider_delivery_partnership ||= delivery_partner.lead_provider_delivery_partnerships.find_by(active_lead_provider:, delivery_partner:)
+    end
+
+    def lead_provider_delivery_partnership_exists
+      return unless lead_provider && delivery_partner
+
+      errors.add(:delivery_partner_api_id, "The entered delivery partner is not recognised to be working in partnership with you for the given cohort. Contact the DfE for more information.") unless lead_provider_delivery_partnership
+    end
+
+    def metadata
+      return unless school && contract_period
+
+      @metadata ||= school.contract_period_metadata.find_by(contract_period:)
+    end
+
+    def not_school_led
+      return unless metadata&.induction_programme_choice == "school_led"
+
+      errors.add(:induction_programme_choice, "The school you have entered has not yet confirmed they will deliver DfE-funded training. Contact the school for more information.")
```

Comment:
new message: `This school has only registered school-led participants. Contact the school for more information.`